core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -47,32 +136,189 @@ public DynamoAdmin(DynamoConfig config) {
+    if (!metadata.getClusteringKeyNames().isEmpty()) {
+      requestBuilder.keySchema(
+          KeySchemaElement.builder().attributeName(CLUSTERING_KEY).keyType(KeyType.RANGE).build());
+    }
+
+    // build local indexes that corresponding to clustering keys
+    if (!metadata.getClusteringKeyNames().isEmpty()) {
+      for (String clusteringKey : metadata.getClusteringKeyNames()) {
+        LocalSecondaryIndex.Builder localSecondaryIndexBuilder =
+            LocalSecondaryIndex.builder()
+                .indexName(getLocalIndexName(namespace, table, clusteringKey))
+                .keySchema(
+                    KeySchemaElement.builder()
+                        .attributeName(clusteringKey)
+                        .keyType(KeyType.HASH)
+                        .build(),
+                    KeySchemaElement.builder()
+                        .attributeName(DynamoOperation.CLUSTERING_KEY)
+                        .keyType(KeyType.RANGE)
+                        .build())
+                .projection(Projection.builder().projectionType(ProjectionType.ALL).build());
+        requestBuilder.localSecondaryIndexes(localSecondaryIndexBuilder.build());
+      }
+    }
+
+    // build secondary indexes
+    if (!metadata.getSecondaryIndexNames().isEmpty()) {
+      for (String secondaryIndex : metadata.getSecondaryIndexNames()) {
+        GlobalSecondaryIndex.Builder globalSecondaryIndexBuilder =
+            GlobalSecondaryIndex.builder()
+                .indexName(getGlobalIndexName(namespace, table, secondaryIndex))
+                .keySchema(
+                    KeySchemaElement.builder()
+                        .attributeName(secondaryIndex)
+                        .keyType(KeyType.HASH)
+                        .build())
+                .projection(Projection.builder().projectionType(ProjectionType.ALL).build());
+        Long ru = DEFAULT_RU;
+        if (options.containsKey(REQUEST_UNIT)) {
+          ru = Long.parseLong(options.get(REQUEST_UNIT));
+        }
+        globalSecondaryIndexBuilder.provisionedThroughput(
+            ProvisionedThroughput.builder().readCapacityUnits(ru).writeCapacityUnits(ru).build());
+        requestBuilder.globalSecondaryIndexes(globalSecondaryIndexBuilder.build());
+      }
+    }
+
+    // ru
+    Long ru = DEFAULT_RU;
+    if (options.containsKey(REQUEST_UNIT)) {
+      ru = Long.parseLong(options.get(REQUEST_UNIT));
+    }
+    requestBuilder.provisionedThroughput(
+        ProvisionedThroughput.builder().readCapacityUnits(ru).writeCapacityUnits(ru).build());
+
+    // table name
+    requestBuilder.tableName(Utility.getFullTableName(namespacePrefix, namespace, table));
+
+    // scaling control
+    boolean noScaling = DEFAULT_NO_SCALING;
+    if (options.containsKey(NO_SCALING)) {
+      if (Boolean.parseBoolean(options.get(NO_SCALING))) {
+        noScaling = true;
+      }
+    }
+    if (!noScaling) {
```

Comment:
Looks like this check is done twice here and in the `controlAutoScaling` method.